cpp/include/cuspatial/detail/distance/linestring_polygon_distance.cuh

```diff
@@ -45,36 +44,44 @@
 namespace cuspatial {
 namespace detail {
 
+/// Device functor that returns true if any of the geometry is empty.
+struct any_input_is_empty {
+  template <typename LhsType, typename RhsType>
+  bool __device__ operator()(LhsType lhs, RhsType rhs)
+  {
+    return lhs.is_empty() || rhs.is_empty();
+  }
+};
+
 /**
  * @brief Computes distances between the multilinestring and multipolygons
  *
- * @param multilinestrings Range to the multilinestring
- * @param multipolygons Range to the multipolygon
+ * This is a load balanced distance compute kernel. Each thread compute exactly 1 pair of segments
```

Comment:
Have you measured performance benefits of the load balancing?

Reply:
Will do some benchmarking to measure it. I can imagine when a kernel is extremely skewed to make the loop long there's performance benefit, but should probably also measure average cases.